From 6be48c9f6a32cdcab9824aaad0c19f92222633c4 Mon Sep 17 00:00:00 2001
From: Kienan Stewart <kstewart@efficios.com>
Date: Tue, 26 Sep 2023 14:45:09 -0400
Subject: [PATCH] fix: lttng-probe-kvm-x86-mmu build with linux 6.6

A small change was made upstream in `spte.h` that requires
`arch/x86/kvm` to be added to the search path when
building lttng-probe-kvm.x86-mmu.o.

See upstream commit :

  commit d10f3780bc2f80744d291e118c0c8bade54ed3b8
  Author: Sean Christopherson <seanjc@google.com>
  Date:   Tue Aug 8 15:40:59 2023 -0700

      KVM: x86/mmu: Include mmu.h in spte.h

      Explicitly include mmu.h in spte.h instead of relying on the "parent" to
      include mmu.h.  spte.h references a variety of macros and variables that
      are defined/declared in mmu.h, and so including spte.h before (or instead
      of) mmu.h will result in build errors, e.g.

Signed-off-by: Kienan Stewart <kstewart@efficios.com>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Change-Id: I5c3fc87d3b006cefbcca198e6e15868a342cb8dd
---
 src/probes/Kbuild | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/probes/Kbuild b/src/probes/Kbuild
index eb0de4a8..73a15844 100644
--- a/src/probes/Kbuild
+++ b/src/probes/Kbuild
@@ -63,7 +63,8 @@ ifneq ($(CONFIG_KVM),)
           ifeq ($(kvm_dep_emulate_check),ok)
             CFLAGS_lttng-probe-kvm-x86.o += -I$(srctree)/virt/kvm \
                        -I$(srctree)/arch/x86/kvm
-            CFLAGS_lttng-probe-kvm-x86-mmu.o += -I$(srctree)/virt/kvm
+            CFLAGS_lttng-probe-kvm-x86-mmu.o += -I$(srctree)/virt/kvm \
+                       -I$(srctree)/arch/x86/kvm
             obj-$(CONFIG_LTTNG) += lttng-probe-kvm-x86.o
             obj-$(CONFIG_LTTNG) += lttng-probe-kvm-x86-mmu.o
           else # ($(kvm_dep_emulate_check),ok)
